contracts/distribution/ZNSRegistrar.sol

```diff
@@ -143,13 +185,23 @@ contract ZNSRegistrar is AccessControlled, UUPSUpgradeable, IZNSRegistrar {
     function revokeDomain(bytes32 domainHash)
     external
     override
-    onlyNameOwner(domainHash)
-    onlyTokenOwner(domainHash)
     {
+        require(
+            isOwnerOf(domainHash, msg.sender, OwnerOf.BOTH),
+            "ZNSRegistrar: Not the owner of both Name and Token"
+        );
```

Comment:
with the if-else we are only running the code inside the block that matches instead of running both checks for every case.
since this function is public, it can be called from any contract to validate the ownership quicker rather than creating checks for every contract. instead, we use Registrar as the checker.

Reply:
what storage slot for BOTH are you talking about? this is a memory var.